homework-submission/week1/src/stores/todosStore.js

```diff
@@ -0,0 +1,166 @@
+import { observable, action, configure, runInAction } from 'mobx';
+
+configure({
+  enforceActions: 'observed'
+});
+
+class TodosStore {
+
+  @observable todos = [];
+  @observable editing = -1;
+  @observable state = 'loading';
+
+  @action async checkBoxHandler(id) {
+    this.state = 'loading';
```

Comment:
I am not sure if you need this here in a small task, that makes the user thinks your application have a kind of lagging, or refreshing while yours is just an (SPA) single page app.